node/contract_test.go

```diff
@@ -0,0 +1,32 @@
+package node
+
+import (
+	"testing"
+
+	"github.com/harmony-one/harmony/consensus"
+	"github.com/harmony-one/harmony/internal/utils"
+	"github.com/harmony-one/harmony/p2p"
+	"github.com/harmony-one/harmony/p2p/p2pimpl"
+)
+
+func prepareNode(t *testing.T) *Node {
+	_, pubKey := utils.GenKey("1", "2")
+	leader := p2p.Peer{IP: "127.0.0.1", Port: "8882", ConsensusPubKey: pubKey}
+	validator := p2p.Peer{IP: "127.0.0.1", Port: "8885"}
+	priKey, _, _ := utils.GenKeyP2P("127.0.0.1", "9902")
+	host, err := p2pimpl.NewHost(&leader, priKey)
+	if err != nil {
+		t.Fatalf("newhost failure: %v", err)
+	}
+	consensus := consensus.New(host, "0", []p2p.Peer{leader, validator}, leader)
+	return New(host, consensus, nil)
+
+}
+
+func TestAddLotteryContract(t *testing.T) {
+	node := prepareNode(t)
+	node.AddLotteryContract()
+	if len(node.DemoContractAddress) == 0 {
```

Comment:
This would test whether the DemoContractAddress has been set. This would be testing L133 `node/contract.go` However we would not know whether the contract gets correctly deployed and whether we are able to call functions within the contract e.g. How to test whether `getPlayers()` works. This would require us to simulate block formation and deployment of contract, you can see how this done here: `test/chain/main.go`.

Reply:
i'll fix this

Reply:
i already tested in local but i didn't write test out. I agree it's not fully tested but it's in WIP - i would add more tests later but at least there wasn't any tests before this PR.

I'm looking to move all the code test/main.go as unit test here